apps/web/src/components/content-management/care-activities/bulk-upload-modal.tsx

```diff
@@ -0,0 +1,172 @@
+import { useHttp } from '@services';
+import { Dispatch, SetStateAction, useState } from 'react';
+import { API_ENDPOINT } from 'src/common';
+import { OccupationItemProps } from 'src/common/interfaces';
+import { Button } from 'src/components/Button';
+import { ModalWrapper } from 'src/components/Modal';
+import {
+  addLegendWorksheet,
+  conditionalFormattingRules,
+  createNewWorkbook,
+  emptyGrayConditionalFormattingRules,
+  getExcelColumnName,
+  HeaderColumns,
+  headerStyle,
+  triggerExcelDownload,
+} from 'src/utils/excel-utils';
+
+interface BulkUploadModalCMSProps {
+  showModal: boolean;
+  setShowModal: Dispatch<SetStateAction<boolean>>;
+}
+
+export const BulkUploadModalCMS: React.FC<BulkUploadModalCMSProps> = ({
+  showModal,
+  setShowModal,
+}) => {
+  const { fetchData } = useHttp();
+  const [isDownloading, setIsDownloading] = useState(false);
+
+  const createUploadTemplate = async (occupations: OccupationItemProps[]) => {
+    const workbook = createNewWorkbook();
+
+    // add upload worksheet
+    const gapMatrixWorksheet = workbook.addWorksheet('Upload_Gap_Matrix', {
+      views: [{ state: 'frozen', ySplit: 1 }],
+    });
+
+    // header row style
+    gapMatrixWorksheet.getRow(1).style = headerStyle;
+
+    // id [readonly] column style
+    gapMatrixWorksheet.getColumn(1).style = headerStyle;
+
+    const columns: HeaderColumns[] = [
+      { header: 'ID (Read only)', key: 'id', width: 6 },
+      { header: 'Care setting', key: 'care_setting', width: 13 },
+      { header: 'Care activity bundle', key: 'care_activity_bundle', width: 13 },
+      { header: 'Care activity', key: 'care_activity', width: 50 },
+      { header: 'Aspect of practice', key: 'aspect_of_practice', width: 13 },
+    ];
+
+    occupations.forEach(occupation => {
+      columns.push({
+        header: occupation.displayName,
+        key: occupation.displayName,
+        width: 13,
+      });
+    });
+
+    // Determine the maximum row and column count
+    const maxRow = 10000;
+    const maxCol = columns.length;
+    const excelMaxRows = 1048576;
+
+    // Add header
+    gapMatrixWorksheet.columns = columns;
+
+    // add conditional formatting rules
+    gapMatrixWorksheet.addConditionalFormatting({
+      ref: `${getExcelColumnName(6)}2:${getExcelColumnName(maxCol)}${maxRow}`,
+      rules: conditionalFormattingRules,
+    });
+
+    // subsequent rows - empty
+    gapMatrixWorksheet.addConditionalFormatting({
+      ref: `A${maxRow + 1}:XFD${excelMaxRows}`,
+      rules: emptyGrayConditionalFormattingRules,
+    });
+
+    // subsequent columns - empty
+    gapMatrixWorksheet.addConditionalFormatting({
+      ref: `${getExcelColumnName(maxCol + 1)}2:XFD${excelMaxRows}`,
+      rules: emptyGrayConditionalFormattingRules,
+    });
+
+    /** Enable sheet protection */
+
+    // Set the locked property of all cells to false (unprotected)
+    for (let rowNumber = 1; rowNumber <= maxRow; rowNumber++) {
```

Comment:
This is just a NIT, perhaps this could be moved to the excel-utils.ts file since it has an initial setup utility. Let me know what you think.

Reply:
The reason I did not move this piece of code to utils is because it is very specific to this excel - with header row and first column locked - which may not always be the case when creating other excels in future.